### DISCUSSION TITLE
influxfb_client

### DISCUSSION THREAD

**Discussion by tutan-chromium** [jertel/vuegraf#84] [General]
Hi,

Trying to follow the instruction and install vuegraf to my BeagleBone Black device, but can't install both influxdb and influxdb-client. First I had to upgrade my outdated influxdb server and it required to uninstall influxdb-client. I can't install influxdb-client after upgrading influxdb to 1.8.10:

```
$ sudo apt install influxdb-client
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  bb-beaglebone-io-installer bb-johnny-five-installer
  ti-c6000-cgt-v8.2.x-installer
Use 'sudo apt autoremove' to remove them.
The following NEW packages will be installed:
  influxdb-client
0 upgraded, 1 newly installed, 0 to remove and 1 not upgraded.
1 not fully installed or removed.
Need to get 1,005 kB of archives.
After this operation, 3,671 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian-security stretch/updates/main armhf influxdb-client armhf 1.1.1+dfsg1-4+deb9u1 [1,005 kB]
Fetched 1,005 kB in 0s (4,001 kB/s)   
Selecting previously unselected package influxdb-client.
(Reading database ... 69428 files and directories currently installed.)
Preparing to unpack .../influxdb-client_1.1.1+dfsg1-4+deb9u1_armhf.deb ...
Unpacking influxdb-client (1.1.1+dfsg1-4+deb9u1) ...
dpkg: error processing archive /var/cache/apt/archives/influxdb-client_1.1.1+dfsg1-4+deb9u1_armhf.deb (--unpack):
 trying to overwrite '/usr/bin/influx', which is also in package influxdb 1.8.10-1
dpkg-deb: error: subprocess paste was killed by signal(Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/influxdb-client_1.1.1+dfsg1-4+deb9u1_armhf.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Am I doing anything wrong?

Thank you!

**Comment by jertel** [comment]
You might not need to install a client package. Is something specific to Vuegraf not working correctly?

**Reply by tutan-chromium** [reply]
I'm confused about dependencies, it explicitly imports `influxdb_client`, but it can't be installed. I can't run vuegraf:
```
$ python3 src/vuegraf/vuegraf.py vuegraf.json
Traceback (most recent call last):
  File "src/vuegraf/vuegraf.py", line 15, in <module>
    import influxdb_client
ImportError: No module named 'influxdb_client'
```
And if i try to install influxdb_client:
```
$ pip3 install influxdb_client
Collecting influxdb_client
  Could not find a version that satisfies the requirement influxdb_client (from versions: )
No matching distribution found for influxdb_client
```

**Reply by jertel** [reply]
Double check that you're on Python 3.9 and pip is updated. Then run the pip install with the `-vvv` parameter to get more details.

You might need to check in with the BeagleBone support forums since I don't have access to a BeagleBone to reproduce this.

**Reply by tutan-chromium** [reply]
Thank you for the reply! I have Python 3.5.3 installed and there is no easy way to upgrade it to 3.9. Is it supposed to work with version <3.9?, like Python 3.7? I mean what is the minimum version required? I'm not a python person, so, I'm a bit confused. I can try to build arm7 version of python3 from sources, but want to be sure what are requirements. Thank you!

**Reply by jertel** [reply]
I've only tested it with python 3.9, so no, it's not _supposed_ to work with python 3.7. But it's possible that it will. Your best bet is to ask the beagle bone community for their advice.
